docker-images/zrok-private-access/docker-compose.yml

```diff
@@ -0,0 +1,64 @@
+version: '3'
+services:
+  zrok-enable-init:
+    image: busybox
+    # matches uid:gid of "nobody" in zrok container image
+    command: chown -Rc 65534:65534 /mnt/.zrok
+    user: root
+    volumes:
+      - zrok_env:/mnt/.zrok
+  zrok-enable:
+    image: docker.io/openziti/zrok:enable-headless  # FIXME: resume :latest if >= :0.3.2
+    depends_on:
+      zrok-enable-init:
+        condition: service_completed_successfully
+    entrypoint:
+      - bash
+      - -c
+      - |
+        if [[ -s /mnt/.zrok/environment.json ]]; then
+          echo "INFO: noop: zrok environment is already enabled"
+          exit 0
+        else
+          echo "INFO: running: zrok $$(sed -E 's/\S{12,}/************/' <<< $${@})"
+          exec zrok $${@}
+        fi
+    command: -- enable --headless ${ZROK_ENABLE_TOKEN}
+    volumes:
+      - zrok_env:/mnt/.zrok
+    environment:
+      HOME: /mnt
+      ZROK_ENABLE_TOKEN:
+      ZROK_API_ENDPOINT: https://api.zrok.io/
+  zrok-private-access:
+    image: docker.io/openziti/zrok
+    command: access private --bind 0.0.0.0:9191 ${ZROK_ACCESS_TOKEN}
+    depends_on:
+      zrok-enable:
+        condition: service_completed_successfully
+    ports:
+      - 9191:9191  # expose the zrok private access proxy to the Docker host
+    stdin_open: true  # FIXME: remove when --headless is available
+    tty: true         # FIXME: remove when --headless is available
+    volumes:
+      - zrok_env:/mnt/.zrok
+    environment:
+      HOME: /mnt
+      PFXLOG_NO_JSON: "true"
+      ZROK_ACCESS_TOKEN:
+
+  # alternatively, access the zrok private access proxy from another container
```

Comment:
alternatively? i don't quite understand the 'why' here

Reply:
This is the private access project and the guide starts you out with a Docker port forward to access the zrok test endpoint. Then at the end of the private share guide it prompts you to try the alternative which is running the demo client app.

Reply:
I'd forgotten to add the final step about running the demo-client, but it's there now.